### PR TITLE
Re-work test suite

### PR DESCRIPTION
**Pull request by labbott**
The existing test suite relied on ITM/ETM for getting test information. We've since built up other tooling that doesn't require those features. Switch to a hiffy based test design to allow for control of tests to happen off target.

### CODE REVIEW

**Comment by labbott** [comment]
See also https://github.com/oxidecomputer/humility/pull/416 for the humility support

File: test/test-api/src/lib.rs [hunk 1]
```diff
@@ -41,10 +41,6 @@ pub enum AssistOp {
 /// Operations that are performed by the test-suite
 #[derive(FromPrimitive)]
 pub enum SuiteOp {
-    /// Get the number of test cases (`() -> usize`).
-    GetCaseCount = 1,
-    /// Get the name of a case (`usize -> [u8]`).
-    GetCaseName = 2,
     /// Run a case, replying before it starts (`usize -> ()`).
     RunCase = 3,
```
Comment (bcantrill) [inline test/test-api/src/lib.rs:45]: Is it worth just eliminating the `SuiteOp` altogether given that there is only one of them now?

File: test/tests-stm32g0/README.mkdn [hunk 2]
```diff
@@ -1,14 +0,0 @@
-# STM32G0 System Tests
```
Comment (bcantrill) [inline test/tests-stm32g0/README.mkdn:1]: Love seeing this go!

**Review by bcantrill — APPROVED** [review]
This looks really great -- thank you for doing this!

File: test/test-api/src/lib.rs (hunk 1 above)
Comment (labbott) [reply test/test-api/src/lib.rs:45]: I thought about that but for consistency I like keeping the enum

**Pull request merged**